airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -620,7 +636,21 @@ private static void validateSchema(final RecordSchemaValidator recordSchemaValid
           validationErrors.put(messageStream, new ImmutablePair<>(updatedErrorMessages, currentCount + 1));
         }
       }
+    }
+  }
 
+  private static void populateUnexpectedFieldNames(AirbyteRecordMessage record, Set<String> fieldsInCatalog, Set<String> unexpectedFieldNames) {
+    final JsonNode data = record.getData();
+    if (data.isObject()) {
+      Iterator<String> fieldNamesInRecord = data.fieldNames();
+      while (fieldNamesInRecord.hasNext()) {
+        final String fieldName = fieldNamesInRecord.next();
+        if (!fieldsInCatalog.contains(fieldName)) {
+          unexpectedFieldNames.add(fieldName);
+        }
+      }
+    } else {
+      throw new RuntimeException(String.format("Unexpected data in record: %s", data));
```

Comment:
is there a posibility to have a valid non object here?

Reply:
I mean it should not happen, but if that happen something is going to break in the code in the `if` block, better to test and throw a more specific exception.
We could use a more specific error message though "Expecting object in record data, got '%s' instead" or something like this.

Reply:
(@malikdiarra you had a comment on the same block)

According to the [Airbyte Protocol](https://docs.airbyte.com/understanding-airbyte/airbyte-protocol/#airbyterecordmessage) the `data` property of a record message must be a JSON of `object` type, so in theory no.

However, on second thought I don't think we should be throwing here in any case - that will basically crash the whole sync, and that's why we have the validation step (which doesn't throw, but rather builds a list of errors).